### PR TITLE
fix #7523 Improvement of error message in case of ambiguous call with…

### LINKED ISSUE #3748: Confusing error message (methods can't have same names as fields if UFCS is used)
```
type
  Argument*[T] = ref object
    defaultValue: T

proc defaultValue(arg: Argument[bool], v: bool): Argument[bool] =
  discard

proc addArgument*(arg: int): Argument[bool] =
  let v = Argument[bool]()
  return v.defaultValue(false)

when isMainModule:
  let quix = addArgument(5)
```

produces

```
foo.nim(12, 11) Error: identifier expected, but found 'v.defaultValue'
```

It took me 20 minutes to figure out what's wrong (in a larger program)

### LINKED ISSUE #7523: Improvement of error message in case of ambiguous call
Condition:

```nim
type
  Foo = object
    bar: int

proc bar(cur: Foo, val: int) =
  discard cur.bar

proc does_fail(): Foo =
  result.bar(5)

can_fail()
```

This results in the following error message:
```
Error: in expression 'result.bar(5)': identifier expected, but found 'result.bar'
```

This error message needs some improvement, because it does not state, that the actual problem is, that there's an ambiguous call to `bar`. 
The compiler cannot know whether the member `bar` of `Foo` is wanted, or whether the function `bar` should be called with `result` (of type `Foo`) as first argument.

### PR DESCRIPTION
**Pull request by bung87**
… field access  
fix #7523
fix #3748

### CODE REVIEW

File: compiler/semcall.nim [hunk 1]
```diff
@@ -416,6 +416,8 @@ proc resolveOverloads(c: PContext, n, orig: PNode,
             let msg = errUndeclaredField % field & " for type " & getProcHeader(c.config, sym)
             localError(c.config, orig[2].info, msg)
         else:
+          if n.kind == nkCall and n[0].kind == nkDotExpr:
+            return result
```
Comment (Araq) [inline compiler/semcall.nim]: It's not easy to see how skipping the error report always ensures that an other, better error is produced.

Comment (bung87) [reply compiler/semcall.nim]: yeah, I thought about that when handle this, it's the only case goes into this branch as far as I know, I thought here may throw error we can handled in outter, but I havn't seen any error throw in compiler code.  I may write code comment there, the chain is bit long.

File: tests/errmsgs/t7523.nim [hunk 2]
```diff
@@ -0,0 +1,13 @@
+discard """
+  action: compile
+  errormsg: "field access 'result.bar' conflicts with procedure call 'bar', rename the field or call as 'bar(result,...)'"
+"""
```
Comment (Araq) [inline tests/errmsgs/t7523.nim:4]: The spec says that `result.bar(5)` is rewritten to `bar(result, 5)` and there is no ambiguity to report.

Comment (bung87) [reply tests/errmsgs/t7523.nim:4]: could u give me the spec link ? am not aware it's in manual or somewhere else. I also discussed with beef ,  beef didn't confirm that behavior is designed.

Comment (Araq) [reply tests/errmsgs/t7523.nim:4]: https://nim-lang.org/docs/manual.html#procedures-method-call-syntax
https://nim-lang.org/docs/manual.html#procedures-properties

Comment (bung87) [reply tests/errmsgs/t7523.nim:4]: It does not make me think as that acorrding to current behavior.  with no conflicts it fit well, with conflict(current implementation exists for years) it act as property access prefered.

Comment (Araq) [reply tests/errmsgs/t7523.nim:4]: Property access does not use `()` though so why would it apply?

Comment (bung87) [reply tests/errmsgs/t7523.nim:4]: that's what I was wondering the first time I face this error.

**Comment by bung87** [comment]
see my next PR https://github.com/nim-lang/Nim/pull/20445

**Pull request closed without merging**